docs/api/saved-objects/bulk_resolve.asciidoc

```diff
@@ -104,13 +104,23 @@ The API returns the following:
 
 Only the index pattern exists, the dashboard was not found.
 
-The `outcome` field may be any of the following:
-
-* `"exactMatch"` -- One document exactly matched the given ID, *or* {kib} failed to find this object.
-* `"aliasMatch"` -- One document with a legacy URL alias matched the given ID; in this case the `saved_object.id` field is different than the given ID.
-* `"conflict"` -- Two documents matched the given ID, one was an exact match and another with a legacy URL alias; in this case the `saved_object` object is the exact match, and the `saved_object.id` field is the same as the given ID.
-
-If the outcome is `"aliasMatch"` or `"conflict"`, the response will also include an `alias_target_id` field. This means that an alias was found for another object, and it describes that other object's ID.
+[NOTE]
+====
+In addition to `saved_object`, several fields can be returned:
+
+* `outcome` (required string) -- One of the following values:
+   - `"exactMatch"` -- One document exactly matched the given ID.
+   - `"aliasMatch"` -- One document with a legacy URL alias matched the given ID; in this case the `saved_object.id` field is different than
+     the given ID.
+   - `"conflict"` -- Two documents matched the given ID, one was an exact match and another with a legacy URL alias; in this case the
+     `saved_object` object is the exact match, and the `saved_object.id` field is the same as the given ID.
+* `alias_target_id` (optional string) -- If the `outcome` is `"aliasMatch"` or `"conflict"`, the response will also include the
+  `alias_target_id` field. This means that an alias was found for another object, and it describes that other object's ID.
+* `suppress_redirect_toast` (optional boolean) -- If the `outcome` is `"aliasMatch"`, the response can also include an optional
```

Comment:
>  in the legacy-url-alias object?

I'd say at least. 

What actual scenario will create these objects with `suppress_redirect_toast: true`? Given `There is currently no way to create them in the Default space organically, though`, I lack some context to suggest a better name...

> Or change the field in the resolve/bulkResolve responses?

Ideally too, but totally optional, I won't force you to re-edit 60ish files just for a better attribute name.

Reply:
Sorry, the context is in #123550.
Right now we are planning to allow the user to choose an option during SO Import to "Preserve URL links between objects" (exact wording TBD).
That will create aliases for any object that has a changed ID, so that weak links between objects still work after import.

I'd prefer to only change the field in the `legacy-url-alias` object, since `resolve/bulkResolve` are instructing the consumer exactly what to do with that information it makes more sense to leave those as-is.

Reply:
Maybe `suppress_redirect_warning` is less UI specific? The warning doesn't need to be a Toast, could be something else theoretically.

It might make sense to invert and always include this property though  instead of including it if the warning should be suppressed: 

```
* alias_redirect_warning (optional boolean) — Included if the outcome is "aliasMatch". Indicates whether a warning should be displayed after redirecting the user to the destination object.
```

This makes it a bit more self explanatory that developers should show a warning after redirecting users and when not to do it. 

I think we can also simplify the description for `alias_target_id` a bit:

```
* alias_target_id (optional string) — Included if the outcome is "aliasMatch". Describes the ID of the destination object.
```

Reply:
Maybe `suppress_redirect_warning` is less UI specific? The warning doesn't need to be a Toast, could be something else theoretically.

It might make sense to invert and always include this property though  instead of including it if the warning should be suppressed: 

```
* alias_redirect_warning (optional boolean) — Included if the outcome is "aliasMatch". Indicates whether a warning should be displayed after redirecting the user to the destination object.
```

This makes it a bit more self explanatory that developers should show a warning after redirecting users and when not to do it. 

I think we can also simplify the description for `alias_target_id` a bit:

```
* alias_target_id (optional string) — Included if the outcome is "aliasMatch". Describes the ID of the destination object.
```

Reply:
Changed to `purpose` in 5f0742d2f7a2afd3356ae9cdd7a1afd7a3b2b404.